lix/source-code/client/src/git/status-list.ts

```diff
@@ -0,0 +1,384 @@
+					return null
+				}
+
+				if ((workdirType === "tree" || workdirType === "special") && !isBlob) {
+					// We don't wan't to include directories in the status list but we still want to walk down there files. - compare (2)
+					return undefined
+				}
+
+				if (stageType === "commit") {
+					// eslint-disable-next-line unicorn/no-null -- commit objects should not be walked, and we stop the walk by returning null - compare (1)
+					return null
+				}
+
+				if ((stageType === "tree" || stageType === "special") && !isBlob) {
+					// We don't wan't to include directories in the status list but we still want to walk down there files. - compare (2)
+					return undefined
+				}
+
+				// Figure out the oids for files, using the staged oid for the working dir oid if the stats match.
+				const headOid = headType === "blob" ? await head?.oid() : undefined
+				const stageOid = stageType === "blob" ? await stage?.oid() : undefined
+
+				let workdirOid
+
+				if (headType !== "blob" && workdirType === "blob" && stageType !== "blob") {
+					workdirOid = addHashes ? await workdir?.oid() : "42" // (use "42" here to avoid hashing of added files - just to realize they are *added - we only expose it for testing purpose)
+				} else if (workdirType === "blob") {
+					workdirOid = await workdir?.oid()
+				}
+
+				const entry: {
+					headOid: string | undefined
+					workdirOid: string | undefined
+					stageOid: string | undefined
+					placeholder?: boolean
+				} = { headOid, workdirOid, stageOid }
+
+				// TODO: 1, 2, 0 > check this meaning? + check if stageid can differ
+				// TODO: only if asked for this file "absent"	file not present in HEAD commit, staging area, or working dir
+
+				// [ 1, 1, 1] "unmodified"
+				if (entry.headOid === entry.workdirOid && entry.workdirOid === entry.stageOid) {
+					if (
+						includeStatus.includes("unmodified") ||
+						filepaths.includes(filepath) ||
+						includeStatus.includes("materialized")
+					) {
+						return [filepath, "unmodified", entry]
+					} else {
+						// the oid is the same and workdir type can not be different from blob (one of the previous if's would have triggered) - filter it out since unchanged
+						return undefined
+					}
+				}
+
+				// [ 0, 2, 0] new, untracked  "*added"	file is untracked, not yet staged
+				if (!entry.headOid && !entry.stageOid && entry.workdirOid) {
+					return [filepath, "*added", entry]
+				}
+
+				// [ 0, 2, 2] added, stage  "added"	previously untracked file, staged
+				if (!entry.headOid && entry.workdirOid && entry.workdirOid === entry.stageOid) {
+					return [filepath, "added", entry]
+				}
+
+				// [ 0, 2, 3] added, staged, (with unstaged changes not exposed in string)
+				if (
+					!entry.headOid &&
+					entry.stageOid &&
+					entry.workdirOid &&
+					entry.workdirOid !== entry.stageOid
+				) {
+					return [filepath, "*added", entry]
```

Comment:
Why do we map this to the same state as 
`// [ 0, 2, 0] new, untracked "*added" file is untracked, not yet staged` previously?